allennlp/data/dataset_readers/t5.py

```diff
@@ -0,0 +1,43 @@
+from typing import Dict
+
+from overrides import overrides
+
+from allennlp.common.file_utils import cached_path
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import TextField, Field
+from allennlp.data.instance import Instance
+from allennlp.data.tokenizers import PretrainedTransformerTokenizer
+from allennlp.data.token_indexers import PretrainedTransformerIndexer
+
+
+@DatasetReader.register("t5")
+class T5DatasetReader(DatasetReader):
+    def __init__(self, model_name: str, **kwargs) -> None:
+        super().__init__(
+            manual_distributed_sharding=True, manual_multiprocess_sharding=True, **kwargs
+        )
+        self.tokenizer = PretrainedTransformerTokenizer(model_name)
+        self.token_indexers = {"tokens": PretrainedTransformerIndexer(model_name)}
+
+    @overrides
+    def _read(self, file_path):
+        with open(cached_path(file_path)) as data_file:
+            for line in self.shard_iterable(data_file):
+                source, target = line.strip().split("\t")
+                yield self.text_to_instance(source, target)
```

Comment:
As of right now, this reader is really designed for fine-tuning tasks, not LM-ing. Should we also have a reader for LM-ing? Or just have a flag that makes it work for LM-ing?

Reply:
The main difference would be that we'd just have source text/tokens, no target.